torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -72,15 +72,29 @@ def create_runtime_wrapper(
     if not hasattr(compiled_fn, "_boxed_call"):
         compiled_fn = make_boxed_func(compiled_fn)
 
+    # args that are still used after we run compiled function (maybe w/ steal_args)
+    epilogue_args_idx = []
+    epilogue_args_idx.extend(runtime_metadata.mutated_inp_runtime_indices)
```

Comment:
It looks like the idea here is that AOTAutograd's runtime epilogue needs to use the input tensors after the compiled fn finishes, so now that we are doing list clearing here we need to manually keep a reference to those inputs?

It does feel pretty fragile... although at least if we get it wrong, it should error loudly (the runtime epilogue will try to access an index not in `orig_inputs`).

Do you mind adding a longer comment describing the situation here? Something like:
```
Note [Inputs needed in runtime epilogue after list clearing]
Compiled Autograd requires us to pass inputs around in a list so that activations can be freed early.
However, we cannot blindly clear the entire list, because AOTAutograd may need access to some of the graph inputs **after** the compiled function has finished running. There are two main cases:
(1) input mutations. If there are an input mutations that we must run outside of the graph, we need access to the input
(2) output aliasing: outputs that aliases graph inputs generally must be regenerated outside of the `autograd.Function`, and doing so requires us accessing the corresonding input after the compiled artifact has run.
```